### ISSUE TITLE
Refactor `TogglSyncManagerFactory`

### ISSUE THREAD

**Issue by simonrozsival** [toggl/mobileapp#3655]
I have several issues with this thing:
- it is too long
- there is so much noise in the code

My suggestion:
- split it into 4 separate files/classes
   - setting up pull transitions
   - setting up push transitions
   - setting up clean-up transitions
   - creating the actual sync manager
- do not write `transitions.ConfigureTransition(` for each transition
   - instead I would suggest using `IDictionary<IStateResult, IState>` and the initializer syntax:
```csharp
var pullTransitons = new Dictionary<IStateResult, IState>
{
    // ...
    [persistPreferences.FinishedPersisting] = persistTags,
    // ...
};
```

What do you think? Let's first have a discussion and agree on a good way of refactoring this mess before we start implementing this refactoring.

Also it will be crucial to make sure we don't mess any transitions in the process. I suggest making sure the generated graphviz representation of the graph stays the same before and after the refactoring.
